Cesium3DTilesContent/src/CmptToGltfConverter.cpp

```diff
@@ -88,26 +90,26 @@ GltfConverterResult CmptToGltfConverter::convert(
           "Composite tile does not contain any loadable inner "
           "tiles.");
     }
-
-    return result;
-  }
-
-  if (innerTiles.size() == 1) {
-    return std::move(innerTiles[0]);
-  }
-
-  for (size_t i = 0; i < innerTiles.size(); ++i) {
-    if (innerTiles[i].model) {
-      if (result.model) {
-        result.model->merge(std::move(*innerTiles[i].model));
-      } else {
-        result.model = std::move(innerTiles[i].model);
-      }
-    }
-
-    result.errors.merge(innerTiles[i].errors);
+    return assetFetcher.asyncSystem.createResolvedFuture(std::move(result));
   }
 
-  return result;
+  return assetFetcher.asyncSystem.all(std::move(innerTiles))
+      .thenImmediately([](std::vector<GltfConverterResult>&& innerResults) {
+        if (innerResults.size() == 1) {
+          return innerResults[0];
```

Comment:
I believe I had the std:move() in there and got a warning about it inhibiting optimization. The code is different than before because innerResults is an rvalue reference and not a local variable. I couldn't tell you if that actually makes a difference :smile: